sdk/program/src/message.rs

```diff
@@ -345,9 +345,14 @@ impl Message {
         self.program_position(i).is_some()
     }
 
-    pub fn is_writable(&self, i: usize) -> bool {
-        i < (self.header.num_required_signatures - self.header.num_readonly_signed_accounts)
-            as usize
+    pub fn is_writable(&self, i: usize, demote_sysvar_write_locks: bool) -> bool {
```

Comment:
From discord chat, I was assuming you do not want to change `is_writable`. Has that changed?

Reply:
Yea. I'm thinking we want to change that now.

Reply:
Because if you could change the lamports without the write lock, then multiple threads doing that might produce inconsistent results.

Reply:
yes, that makes sense